### PR TITLE
CSPL-1801 - Avoid odd number of arguments passed for logging

### PR DESCRIPTION
**Pull request by mgaldino-splunk**
This issue is most likely never exposed, but it can cause the Operator to crash under certain conditions. 

The third parameter for `scopedLog.Error()` should be a set of Key and values, as per the function declaration:

**go/pkg/mod/github.com/go-logr/logr@v1.2.0/logr.go:261**
```
func (l Logger) Error(err error, msg string, keysAndValues ...interface{}) {

}
```

When only one value is passed we can hit this crash for `odd number of arguments passed as key-value pairs for logging`

```
1.652302394894617e+09	DPANIC	controller.indexercluster.indexerClusterPodManager.getClusterManagerClient odd number of arguments passed as key-value pairs for logging	{"reconciler group": "enterprise.splunk.com", "reconciler kind": "IndexerCluster", "name": "example-idc", "namespace": "default", "ignored key": "Couldn't find pod"}
```

Sample Stack:
```
github.com/splunk/splunk-operator/pkg/splunk/enterprise.(*indexerClusterPodManager).verifyRFPeers
	/workspace/pkg/splunk/enterprise/indexercluster.go:793
github.com/splunk/splunk-operator/pkg/splunk/enterprise.glob..func2
	/workspace/pkg/splunk/enterprise/indexercluster.go:422
github.com/splunk/splunk-operator/pkg/splunk/enterprise.ApplyIndexerCluster
	/workspace/pkg/splunk/enterprise/indexercluster.go:304
github.com/splunk/splunk-operator/controllers.glob..func3
	/workspace/controllers/indexercluster_controller.go:112
github.com/splunk/splunk-operator/controllers.(*IndexerClusterReconciler).Reconcile
	/workspace/controllers/indexercluster_controller.go:104
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227
```

### CODE REVIEW

**Comment by sgontla** [comment]
I think it's high time for us to implement a wrapper, that does the sanity on the number arguments, and use that wrapper all over the place. Probably, through a separate Jira.

File: pkg/splunk/enterprise/indexercluster.go [hunk 1]
```diff
@@ -573,7 +573,7 @@ func (mgr *indexerClusterPodManager) getClusterManagerClient(ctx context.Context
 	podName := fmt.Sprintf(splcommon.TestClusterManagerID, managerIdxcName, "0")
 	adminPwd, err := splutil.GetSpecificSecretTokenFromPod(ctx, mgr.c, podName, mgr.cr.GetNamespace(), "password")
 	if err != nil {
-		scopedLog.Error(err, "Couldn't retrieve the admin password from pod %v", err.Error())
+		scopedLog.Error(err, "Couldn't retrieve the admin password from pod %v", "podName", podName)
```
[the remark this answers is not in the document]
Comment (akondur) [reply pkg/splunk/enterprise/indexercluster.go:576]: +1

**Review by akondur — APPROVED** [review]

File: pkg/splunk/enterprise/indexercluster.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgaldino-splunk) [reply pkg/splunk/enterprise/indexercluster.go]: Updated with `fmt.Sprintf` instead so we don't lose information when troubleshooting. Please let me know what you think.

File: pkg/splunk/client/awss3client.go [hunk 2]
```diff
@@ -249,7 +249,7 @@ func (awsclient *AWSS3Client) DownloadApp(ctx context.Context, remoteFile, local
 			IfMatch: aws.String(etag),
 		})
 	if err != nil {
-		scopedLog.Error(err, "Unable to download item %s, %v", remoteFile, err)
+		scopedLog.Error(err, fmt.Sprintf("Unable to download item %s, %v", remoteFile, err))
```
Comment (jryb) [inline pkg/splunk/client/awss3client.go:252]: The `err` is already included in the log message so we can probably remove it from the `fmt.Sprintf()`.

File: pkg/splunk/enterprise/indexercluster.go [hunk 3]
```diff
@@ -573,7 +573,7 @@ func (mgr *indexerClusterPodManager) getClusterManagerClient(ctx context.Context
 	podName := fmt.Sprintf(splcommon.TestClusterManagerID, managerIdxcName, "0")
 	adminPwd, err := splutil.GetSpecificSecretTokenFromPod(ctx, mgr.c, podName, mgr.cr.GetNamespace(), "password")
 	if err != nil {
-		scopedLog.Error(err, "Couldn't retrieve the admin password from pod %v", err.Error())
+		scopedLog.Error(err, fmt.Sprintf("Couldn't retrieve the admin password from pod %v", err.Error()))
```
Comment (jryb) [inline pkg/splunk/enterprise/indexercluster.go:576]: Same here.  Just remove the `%v` and `err.Error()`

**Review by jryb — APPROVED** [review]

File: pkg/splunk/client/awss3client.go (hunk 2 above)
Comment (mgaldino-splunk) [reply pkg/splunk/client/awss3client.go]: Good point. Resolved in the last commit.

File: pkg/splunk/enterprise/indexercluster.go (hunk 3 above)
Comment (mgaldino-splunk) [reply pkg/splunk/enterprise/indexercluster.go]: I think this one was intended to be `podName`. Fixed in the latest commit.

**Review by vivekbasappa — APPROVED** [review]

**Review by vivekr-splunk — APPROVED** [review]

**Review by sgontla — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Pull request merged**